GHC 8.4 compilation failure (SMP-related)

```
[5 of 5] Compiling LambdaCube.PipelineSchemaUtil ( src/LambdaCube/PipelineSchemaUtil.hs, dist/build/LambdaCube/PipelineSchemaUtil.o )

src/LambdaCube/PipelineSchemaUtil.hs:17:10: error:
    • No instance for (Semigroup PipelineSchema)
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Monoid PipelineSchema’
   |
17 | instance Monoid PipelineSchema where
   |          ^^^^^^^^^^^^^^^^^^^^^
```

--- a/lambdacube-ir.haskell/lambdacube-ir.cabal
+++ b/lambdacube-ir.haskell/lambdacube-ir.cabal
@@ -25,11 +25,11 @@ library
   -- other-modules:       
   other-extensions:    OverloadedStrings, RecordWildCards, DeriveFunctor
   -- CAUTION: When the build-depends change, please bump the git submodule in lambdacube-docker repository
-  build-depends:       base >=4.9 && <4.10,
-                       containers >=0.5 && <0.6,
-                       vector >=0.12 && <0.13,
-                       text >=1.2 && <1.3,
-                       aeson >=1.1 && <1.2,
-                       mtl >=2.2 && <2.3
+  build-depends:       base >=4.9,
+                       containers >=0.5,
+                       vector >=0.12,
+                       text >=1.2,
+                       aeson >=1.1,
+                       mtl >=2.2
   hs-source-dirs:      src
   default-language:    Haskell2010

--- a/lambdacube-ir.haskell/src/LambdaCube/PipelineSchemaUtil.hs
+++ b/lambdacube-ir.haskell/src/LambdaCube/PipelineSchemaUtil.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE FlexibleContexts, TypeSynonymInstances, FlexibleInstances #-}
+{-# LANGUAGE CPP, FlexibleContexts, TypeSynonymInstances, FlexibleInstances #-}
 module LambdaCube.PipelineSchemaUtil where
 
 import Control.Monad.Writer
@@ -16,5 +16,11 @@ unionObjectArraySchema (ObjectArraySchema a1 b1) (ObjectArraySchema a2 b2) =
 
 instance Monoid PipelineSchema where
   mempty = PipelineSchema mempty mempty
+#if !MIN_VERSION_base(4,11,0)
   mappend (PipelineSchema a1 b1) (PipelineSchema a2 b2) =
     PipelineSchema (Map.unionWith unionObjectArraySchema a1 a2) (Map.unionWith (\a b -> if a == b then a else error $ "schema type mismatch " ++ show (a,b)) b1 b2)
+#else
+instance Semigroup PipelineSchema where
+  (<>)    (PipelineSchema a1 b1) (PipelineSchema a2 b2) =
+    PipelineSchema (Map.unionWith unionObjectArraySchema a1 a2) (Map.unionWith (\a b -> if a == b then a else error $ "schema type mismatch " ++ show (a,b)) b1 b2)
+#endif

